apps/one-drive/src/inngest/functions/data-protection/sync-items.ts

```diff
@@ -0,0 +1,302 @@
+  }
+  return chunks;
+};
+
+export const formatPermissions = (
+  permission: MicrosoftDriveItemPermissions
+): DataProtectionObjectPermission[] | [] => {
+  if (permission.grantedToV2?.user) {
+    return [
+      {
+        id: permission.id,
+        type: 'user',
+        displayName: permission.grantedToV2.user.displayName,
+        userId: permission.grantedToV2.user.id,
+        email: permission.grantedToV2.user.email,
+        metadata: {
+          permissionId: permission.id,
+          roles: permission.roles,
+        },
+      },
+    ];
+  } else if (permission.grantedToIdentitiesV2) {
+    const formattedPermissions: DataProtectionObjectPermission[] = [];
+
+    for (const p of permission.grantedToIdentitiesV2) {
+      if (p.user) {
+        formattedPermissions.push({
+          id: `${permission.id}${env.ID_SEPARATOR}${p.user.id}`,
+          type: 'user',
+          displayName: p.user.displayName,
+          userId: p.user.id,
+          email: p.user.email,
+          metadata: {
+            permissionId: permission.id,
+            roles: permission.roles,
+          },
+        });
+      }
+    }
+
+    return formattedPermissions;
+  }
+  return [];
+};
+
+export const parseDataProtetionItems = (
+  itemsWithPermisions: ItemsWithPermisions[]
+): DataProtectionObject[] => {
+  const dataProtection: DataProtectionObject[] = [];
+
+  for (const { item, permissions } of itemsWithPermisions) {
+    if (item.createdBy.user.id) {
+      const validPermissions: MicrosoftDriveItemPermissions[] = permissions.filter((p) => {
+        if (p.link && p.link.scope === 'users') return true;
+        else if (p.grantedToV2?.user) return true;
+        return false;
+      });
+
+      if (validPermissions.length) {
+        const dataProtectionItem = {
+          id: item.id,
+          name: item.name,
+          ownerId: item.createdBy.user.id,
+          url: item.webUrl,
+          permissions: validPermissions.map((p) => formatPermissions(p)).flat(),
+        };
+
+        dataProtection.push(dataProtectionItem);
+      }
+    }
+  }
+
+  return dataProtection;
+};
+
+export const syncItems = inngest.createFunction(
+  {
+    id: 'one-drive-sync-items',
+    priority: {
+      run: 'event.data.isFirstSync ? 600 : 0',
+    },
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: env.MICROSOFT_DATA_PROTECTION_ITEMS_SYNC_CONCURRENCY,
+    },
+    cancelOn: [
+      {
+        event: 'one-drive/one-drive.elba_app.uninstalled',
+        match: 'data.organisationId',
+      },
+      {
+        event: 'one-drive/one-drive.elba_app.installed',
+        match: 'data.organisationId',
+      },
+    ],
+    retries: env.MICROSOFT_DATA_PROTECTION_SYNC_MAX_RETRY,
+  },
+  { event: 'one-drive/items.sync.triggered' },
+  async ({ event, step }) => {
+    const { siteId, driveId, isFirstSync, folderId, skipToken, organisationId } = event.data;
+
+    logger.info('Sync Items');

```

Comment:
```suggestion
  async ({ event, step, logger }) => {
    const { siteId, driveId, isFirstSync, folderId, skipToken, organisationId } = event.data;

    logger.info('Sync Items');

```

My bad, I forgot to use the inngest logger in other integration and in the template. If it's used outside a step it will print many times.

Reply:
Updated, thanks for advice.. But I think that we can remove it later, it's more for me to see how function goes